src/main/java/com/internship/socialnetwork/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.internship.socialnetwork.service.impl;
+
+import com.internship.socialnetwork.dto.UserDto;
+import com.internship.socialnetwork.exception.NotFoundException;
+import com.internship.socialnetwork.exception.BadRequestException;
+import com.internship.socialnetwork.model.User;
+import com.internship.socialnetwork.repository.UserRepository;
+import com.internship.socialnetwork.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+
+    public List<UserDto> getAllUsers() {
+        return userRepository.findAll().stream().map(UserDto::convertToDto).collect(Collectors.toList());
+    }
+
+    public User getUserModel(Long id) {
+        return userRepository.findById(id).orElseThrow(() -> new NotFoundException("User not found!"));
+    }
+
+    public UserDto getUser(Long id) {
+        return UserDto.convertToDto(getUserModel((id)));
```

Comment:
You can use static import convertToDto since you are using it in multiple places